enterprise/frontend/src/embedding-sdk/conventional-changelog-config.js

```diff
@@ -1,4 +1,51 @@
 // eslint-disable-next-line import/no-commonjs
 module.exports = {
-  gitRawCommitsOpts: { grep: "(sdk)" },
+  gitRawCommitsOpts: {
+    grep: "(sdk)",
+  },
+  options: {
+    preset: "angular",
+    pkg: {
+      path: "enterprise/frontend/src/embedding-sdk/package.template.json",
+    },
+    tagPrefix: "embedding-sdk-",
+    transform: function (commit, cb) {
+      if (typeof commit.gitTags === "string") {
+        const tagPrefix = "embedding-sdk-";
+        const rtag = new RegExp(`tag:\\s*[=]?${tagPrefix}(.+?)[,)]`, "gi");
+        const match = rtag.exec(commit.gitTags);
+        rtag.lastIndex = 0;
+        if (match) {
+          commit.version = match[1];
+        }
+      }
+      if (commit.committerDate) {
+        const dateFormatter = Intl.DateTimeFormat("sv-SE", {
+          timeZone: "UTC",
+        });
+        commit.committerDate = dateFormatter.format(
+          new Date(commit.committerDate),
+        );
+      }
+
+      if (hasBackportPrefix(commit)) {
+        const { groups } =
+          /\"(?<type>\w*)(?:\((?<scope>[\w\$\.\-\* ]*)\))?\: (?<message1>.*)\"(?<message2>.*)$/.exec(
+            commit.header,
+          );
+        commit = {
+          ...commit,
+          type: groups.type,
+          scope: groups.scope,
+          subject: groups.message1 + groups.message2,
+        };
+      }
+
+      cb(null, commit);
+    },
+  },
 };
+
+function hasBackportPrefix(commit) {
+  return commit.header.startsWith("🤖 backported");
```

Comment:
So if you did a manual backport, and you follow the conventional commit it won't be included in the changelog anyway.

This works for the auto backport, but for manual ones it is what it is. If you name your PR `This it a backport of xxxx` it's not gonna be included. You gotta do `fix(sdk): xxxxx`

I did think about the `backport` name, but it doesn't make sense to have that in the changelog, the changelog entries should state what is added to a specific version, it doesn't need to show the code is from a backport PR.

If we do `fix(sdk): [backport] xxx` we should remove the `[backport]` bit from the entry anyway in my opinion.

What scares you about the manual backports?

Reply:
> So if you did a manual backport, and you follow the conventional commit it won't be included in the changelog anyway.

Makes sense, we'll have to keep that in mind, I think my first instinct would be to call them `manual backport of fix(sdk)`.

I still wonder if we should remove anything up to a conventional commit "start" though, ie: `whatever blabla fix(sdk): something` -> `fix(sdk): something`.
That would work for all cases and false positives are very unlikely, at most we'd have reverts in the automatic changelog which is not bad actually.
WDYT?

Reply:
if we structure our message a bit better, it would.

Now the auto backport commit is wrapped inside the quotes `🤖 backported "fix(sdk): Your PR title" (#<backport_pr_number>)`. And what we need to set as a subject in this case is just `fix(sdk): Your PR title`.

This makes extracting the actual message a bit harder since it will also get the the part after the quote.

I have a test that matches starting from `fix(sdk): ` but it also includes the last quote from the backport PR, which looks really off. If we can live with that dangling quote then we could adjust this regex.

Reply:
I've included the above detail in the original issue.